excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/HiddenFieldsValidator.java

```diff
@@ -46,6 +49,38 @@ public Boolean parseComplexTypesHiddenFields(DefinitionDataItem definitionDataIt
         return retainHiddenValue;
     }
 
+    public Boolean parseCaseEventComplexTypesHiddenFields(DefinitionDataItem definitionDataItem,
+                                                          Map<String, DefinitionSheet> definitionSheets) {
+        if (definitionDataItem.getFieldShowCondition() != null
+            && definitionDataItem.getRetainHiddenValue() != null) {
+            complexTypeHasRetainHiddenValue(definitionDataItem,
+                definitionSheets.get(SheetName.COMPLEX_TYPES.getName()));
+        }
+
+        return definitionDataItem.getRetainHiddenValue();
+    }
+
+
+    private void complexTypeHasRetainHiddenValue(DefinitionDataItem definitionDataItem,
+                                                 DefinitionSheet complexType) {
+        List<DefinitionDataItem> complexTypeList = complexType.getDataItems().stream()
+            .filter(definitionDataItem1 -> definitionDataItem1.getId()
+                .equals(definitionDataItem.getId()) && definitionDataItem1.getListElementCode()
+                .equals(definitionDataItem.getListElementCode())).collect(toList());
+
+        boolean match = complexTypeList.stream().noneMatch(ddi -> {
+            return ddi.getFieldShowCondition() == null
+                && (ddi.getRetainHiddenValue() != null || ddi.getRetainHiddenValue() == null);
```

Comment:
This will always be true?
`(ddi.getRetainHiddenValue() != null || ddi.getRetainHiddenValue() == null)`

Reply:
logic changed after CP, so removed